torch/csrc/autograd/utils/wrap_outputs.h

```diff
@@ -47,6 +48,10 @@ inline PyObject* wrap(THPDtype *dtype) {
   return (PyObject*)dtype;
 }
 
+inline PyObject* wrap(at::ScalarType scalarType) {
```

Comment:
seems like if we have all this we can probably port promoteTypes to native_functions and this would just work.  Are you planning to do that?

Reply:
yes, but in a separate PR